File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;   // negatives are not palindromes
        int orig = x, rev = 0;
        while (x != 0) {
            int digit = x % 10;
            if (rev > (Integer.MAX_VALUE - digit) / 10) return false; // overflow check
            rev = rev * 10 + digit;
            x /= 10;
        }
        return rev == orig;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        org =x
        ans=0
        rem =0
        while x!=0:
            rem = x%10
            ans = ans*10 + rem
            x = x//10
        return ans == org